rmw_zenoh_cpp/src/detail/rmw_client_data.cpp

```diff
@@ -94,13 +81,28 @@ void client_data_handler(z_owned_reply_t * reply, void * data)
   std::chrono::nanoseconds::rep received_timestamp =
     std::chrono::system_clock::now().time_since_epoch().count();
 
-  client_data_shared_ptr->add_new_reply(
+  client_data->add_new_reply(
     std::make_unique<rmw_zenoh_cpp::ZenohReply>(reply, received_timestamp));
 
   // Since we took ownership of the reply, null it out here
   *reply = z_reply_null();
 }
 
+///=============================================================================
+void client_data_drop(void * data)
+{
+  auto client_data = static_cast<rmw_zenoh_cpp::ClientData *>(data);
+  if (client_data == nullptr) {
+    RMW_ZENOH_LOG_ERROR_NAMED(
+      "rmw_zenoh_cpp",
+      "Unable to obtain client_data_t "
```

Comment:
```suggestion
      "Unable to obtain client_data_t from data in client_data_drop. "
```

Reply:
Good idea, fixed in [ec6c181](https://github.com/ros2/rmw_zenoh/pull/310/commits/ec6c181797d9d234f9577de6154ede4067d4b760)